### PR TITLE
fix(escalating): Return forecast for one event if no forecast exists

### LINKED ISSUE #53083: Forecast does not exist when checking if escalating
Forecast does not exist when checking if an issue is escalating if the issue
- has not been seen for 7 days
- nodestore TTL has expired

Tracks this ticket on the bug fix board: https://github.com/getsentry/sentry/issues/52841

### PR DESCRIPTION
**Pull request by jangjodi**
Forecast does not exist when checking if an issue is escalating if the issue:
- has not been seen for 7 days
- nodestore TTL has expired

On fetch, return the forecast for one event if no forecast exists in nodestore. This will be accurate enough until the forecast can be regenerated, as the issue will meet the above criteria and this event will be this issue's first in 7 days.

fixes https://github.com/getsentry/sentry/issues/53083
fixes SENTRY-134Y
fixes SENTRY-11PS

### CODE REVIEW

File: tests/sentry/tasks/test_weekly_escalating_forecast.py [hunk 1]
```diff
@@ -41,7 +45,9 @@ def test_empty_escalating_forecast(
             run_escalating_forecast()
             group = group_list[0]
             fetched_forecast = EscalatingGroupForecast.fetch(group.project.id, group.id)
-            assert fetched_forecast is None
+            assert fetched_forecast is not None
+            if fetched_forecast:
```
Comment (scttcper) [inline tests/sentry/tasks/test_weekly_escalating_forecast.py:49]: not sure the if here is necessary

Comment (scttcper) [reply tests/sentry/tasks/test_weekly_escalating_forecast.py:49]: is it a type thing?

Comment (jangjodi) [reply tests/sentry/tasks/test_weekly_escalating_forecast.py:49]: yeah, typing fails without it

**Review by scttcper — APPROVED** [review]

File: tests/sentry/tasks/test_weekly_escalating_forecast.py (hunk 1 above)
Comment (snigdhas) [reply tests/sentry/tasks/test_weekly_escalating_forecast.py:49]: you could try something like this but what you have is fine! 
```suggestion
            assert fetched_forecast and fetched_forecast.forecast == ONE_EVENT_FORECAST
```

**Review by snigdhas — APPROVED** [review]

**Pull request merged**